src/main/kotlin/com/example/petshopkotlin/pet/PetService.kt

```diff
@@ -0,0 +1,40 @@
+package com.example.petshopkotlin.pet
+
+import com.example.petshopkotlin.pet.model.Pet
+import org.bson.types.ObjectId
+import org.springframework.stereotype.Service
+
+@Service
+class PetService(val repo: PetRepository) {
+    fun getPets() = repo.findAllByOrderByIdDesc()
+
+    fun addPet(pet: Pet): Pet {
+        val errors = validatePet(pet)
+        require(errors.isBlank()) { errors }
+
+        return repo.save(pet)
+    }
+
+    fun adoptPet(id: ObjectId): String {
+
+        val pet = repo.findById(id).orElseThrow { IllegalArgumentException("Pet with ID: $id doesn't exist") }
```

Comment:
Mongo repository is returning a Java Optional. Thus I use` orElseThrow ` method to throw an error if the value doesn't exist. If value exists then it will return Pet.

How can I implement it better?